tinygrad/lazy.py

```diff
@@ -225,9 +225,13 @@ def e(self:LazyBuffer, op:Union[UnaryOps, BinaryOps, TernaryOps], *srcs:LazyBuff
 
     if MERGE_ELEMENTWISE_OPS:
       # remove the buffers from any (childless) BinaryOps that feed into this
-      srcs = tuple([x.op if x.optype == BinaryOps and not x.children and not x.realized else x for x in srcs])  # type: ignore
+      merged_srcs:Tuple[Union[LazyOp, LazyBuffer], ...] = tuple([x.op if x.optype == BinaryOps and not x.children and not x.realized else x for x in srcs])  # type: ignore
+      # NOTE: this is incompete, you can still fuse with reduce ops and exceed the limit
+      merged_srcs = merged_srcs if self.device != "METAL" or sum(len(x.buffers) for x in merged_srcs) < 30 else srcs
```

Comment:
should this be done in a more generic way instead? cuz webgpu also needs a buffer limit of 8.

Reply:
yea, probably. i'm fine with a PR refactoring this and testing it for webgpu. I mostly need to get #2006 back in for openpilot